whereby/sdk
# Add kick participant and end meeting actions for meeting hosts

### Description

Allow participants joining with the `host` role to kick a remote participant and/or end the meeting for all remote participants.

### Testing

1. Run local-stack and join a room with 1 or more participants via PWA.
2. Check out this branch
3. Ensure that STORYBOOK_ROOM is the same URL as the joined room in step 1.
4. In the `/.env` file add a valid _host_ `roomKey` for the configured `STORYBOOK_ROOM` in following line:

```
STORYBOOK_ROOM_HOST_ROOMKEY=[host_roomKey_here]
```

5. Run `yarn build && yarn dev`
6. Open the storybook story: "Custom UI" > "Room Connection With Host Controls" ([direct localhost link](http://localhost:6006/?path=/story/examples-custom-ui--room-connection-with-host-controls))
7. Check that the `roomKey` is filled with the host room key added in step 4.
8. Clicking on the "End meeting" button and individual "Kick" buttons should now perform the respective actions in the connected room [Screenshot 1].
9. In the storybook controls panel, remove the `roomKey` parameter and reload the story
10. The "End meeting" button and individual "Kick" buttons shown in the UI should now be in red [Screenshot 2]
11. Clicking on the "End meeting" button or any of the individual "Kick" buttons should not perform the action in the connected room (not connected as a "host" now).
12. Additionally, a warning should be displayed in the console whenever these actions are invoked by a non-"host" user.

### Screenshots/GIFs (if applicable)

#### Screenshot 1: "host" `roomKey` provided

<img width="520" alt="Screenshot 2024-04-17 at 15 47 04" src="https://github.com/whereby/sdk/assets/119658286/45a2257f-d052-4311-9a98-f4792e9913eb">


#### Screenshot 2: "host" `roomKey` not provided

<img width="554" alt="Screenshot 2024-04-17 at 15 47 17" src="https://github.com/whereby/sdk/assets/119658286/9e05e684-c010-48a4-8d00-a168fa9a5e8f">

### Checklist

        under the terms of the MIT license.
